pandas/core/internals.py

```diff
@@ -4440,7 +4440,10 @@ def _lcd_dtype(l):
         """ find the lowest dtype that can accomodate the given types """
         m = l[0].dtype
         for x in l[1:]:
-            if x.dtype.itemsize > m.itemsize:
+            # the new dtype must either be wider or a strict subtype
+            if (x.dtype.itemsize > m.itemsize or
```

Comment:
doesn't ``np.find_common_type`` do this? (or ``.common_type``)?

should create a pandas version of this to isolate (e.g. this won't handle non-numpy types)